### PR TITLE
Implement MIDP high-level LCDUI screens and items

### PR DESCRIPTION
**Pull request by dlunch**
## Summary
- Implement MIDP Form, Screen, and Alert rendering and lifecycle through the existing Display/Graphics path.
- Add Item layout, ownership, focus, scrolling, ChoiceGroup interaction, StringItem, ImageItem, Gauge, and listener dispatch.
- Implement title, ticker, softkey/menu commands, live viewport updates, and serialized Alert/Ticker timers with guest-backed state.
- Preserve Canvas/GameCanvas behavior and paint-before-callback ordering; use injected platform fonts for measurement and wrapping.
- Add focused inline regression tests and a controllable test clock; document the public-specification/reference-implementation boundary in AGENTS.md.

## Validation
- `cargo test --workspace`
- `cargo clippy --workspace -- -D warnings`
- `cargo clippy --target wasm32-unknown-unknown --workspace --exclude wie-app -- -D warnings`
- `cargo fmt --all -- --check`
- `git diff --check`

Validation uses JVM unit tests and backing-image checks. Frontend visual smoke testing was not performed.

### CODE REVIEW

File: wie-midp/src/classes/net/wie/choice_element.rs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            )],
+            fields: vec![
+                JavaFieldProto::new("text", "Ljava/lang/String;", FieldAccessFlags::empty()),
+                JavaFieldProto::new("image", "Ljavax/microedition/lcdui/Image;", FieldAccessFlags::empty()),
+                JavaFieldProto::new("displayImage", "Ljavax/microedition/lcdui/Image;", FieldAccessFlags::empty()),
+                JavaFieldProto::new("font", "Ljavax/microedition/lcdui/Font;", FieldAccessFlags::empty()),
+                JavaFieldProto::new("selected", "Z", FieldAccessFlags::empty()),
+            ],
+            access_flags: ClassAccessFlags::empty(),
+        }
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/net/wie/choice_element.rs:35]: Checked the jvm 0.1.1 native API used here: new_class invokes instantiate_class and invoke_special without Java package-access checks, and get_field/put_field likewise do not enforce Java visibility. ChoiceElement is an internal guest record accessed through these Rust APIs, not an application bytecode reference. The reported IllegalAccessError does not occur on this path; keeping the helper non-public.

File: wie-midp/src/classes/javax/microedition/lcdui/displayable.rs [hunk 2]
```diff
@@ -18,28 +19,77 @@ impl Displayable {
             parent_class: Some("java/lang/Object"),
             interfaces: vec![],
             methods: vec![
-                JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::PROTECTED),
+                JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::empty()),
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/displayable.rs:52]: Keeping the internal package-private constructor. The public MIDP 2.0 API does not expose a protected Displayable constructor; its sizeChanged documentation explicitly prohibits applications from directly subclassing Displayable. The previous modifier is not the API contract. [Public API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Displayable.html#sizeChanged(int,%20int)).

File: wie-midp/src/classes/javax/microedition/lcdui/item.rs [hunk 3]
```diff
@@ -1,31 +1,952 @@
...
     pub fn as_proto() -> WieJavaClassProto {
         WieJavaClassProto {
             name: "javax/microedition/lcdui/Item",
             parent_class: Some("java/lang/Object"),
             interfaces: vec![],
-            methods: vec![JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::PROTECTED)],
-            fields: vec![],
+            methods: vec![
+                JavaMethodProto::new("<clinit>", "()V", Self::cl_init, MethodAccessFlags::STATIC),
+                JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::empty()),
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/item.rs:37]: Keeping the internal package-private constructor. The public MIDP 2.0 Item API exposes no public or protected constructor. Application-defined item implementations subclass CustomItem, whose constructor is protected; that does not require exposing Item.<init>. [Item API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Item.html), [CustomItem API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/CustomItem.html).

File: wie-midp/src/classes/javax/microedition/lcdui/screen.rs [hunk 4]
```diff
@@ -15,7 +17,15 @@ impl Screen {
             name: "javax/microedition/lcdui/Screen",
             parent_class: Some("javax/microedition/lcdui/Displayable"),
             interfaces: vec![],
-            methods: vec![JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::PROTECTED)],
+            methods: vec![
+                JavaMethodProto::new("<init>", "()V", Self::init, MethodAccessFlags::empty()),
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/screen.rs:21]: Keeping the internal package-private constructor. The public MIDP 2.0 Screen API exposes no public or protected constructor. Its public concrete subclasses provide the application entry points, so restoring the previous protected modifier would not match the public API. [Screen API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Screen.html).

File: wie-midp/src/classes/javax/microedition/lcdui/form.rs [hunk 5]
```diff
@@ -20,31 +32,1878 @@ impl Form {
...
+            if row.y.saturating_add(row.height) <= scroll || row.y >= viewport_bottom {
+                continue;
+            }
+            let _: () = jvm
+                .invoke_virtual(
+                    &row.item,
+                    "javax/microedition/lcdui/Item",
+                    "paintItem",
+                    "(Ljavax/microedition/lcdui/Graphics;IIIIZ)V",
+                    (graphics.clone(), 0, row.y - scroll, width, row.height, index as i32 == focus_index),
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/form.rs]: Fixed in 6d0fce67. Form rows now retain their allocated width and x-position, and use that width consistently for height measurement, painting, and focus bounds. Horizontal alignment and expansion are applied within the existing single-column layout; locked StringItems keep their width as required by MIDP. Added a pixel regression covering wrapping, alignment, and expansion. All 308 workspace tests and workspace Clippy pass.

File: wie-midp/src/classes/javax/microedition/lcdui/displayable.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/displayable.rs:52]: Keeping the internal package-private constructor. The public MIDP 2.0 API does not expose a protected Displayable constructor; its sizeChanged documentation explicitly prohibits applications from directly subclassing Displayable. The previous modifier is not the API contract. [Public API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Displayable.html#sizeChanged(int,%20int)).

File: wie-midp/src/classes/javax/microedition/lcdui/item.rs [hunk 6]
```diff
@@ -1,31 +1,968 @@
...
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/item.rs:37]: Keeping the internal package-private constructor. The public MIDP 2.0 Item API exposes no public or protected constructor. Application-defined item implementations subclass CustomItem, whose constructor is protected; that does not require exposing Item.<init>. [Item API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Item.html), [CustomItem API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/CustomItem.html).

File: wie-midp/src/classes/javax/microedition/lcdui/screen.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/screen.rs:21]: Keeping the internal package-private constructor. The public MIDP 2.0 Screen API exposes no public or protected constructor. Its public concrete subclasses provide the application entry points, so restoring the previous protected modifier would not match the public API. [Screen API](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/Screen.html).

File: wie-midp/src/classes/javax/microedition/lcdui/image_item.rs [hunk 7]
```diff
@@ -0,0 +1,560 @@
...
+        graphics: ClassInstanceRef<Graphics>,
+        x: i32,
+        y: i32,
+        width: i32,
+        height: i32,
+        _focused: bool,
+    ) -> JvmResult<()> {
+        let Some((image, image_width, image_height)) = Self::display_image(jvm, &this).await? else {
+            return Ok(());
+        };
```
[the remark this answers is not in the document]
Comment (dlunch) [reply wie-midp/src/classes/javax/microedition/lcdui/image_item.rs:308]: Keeping the empty-image behavior: MIDP explicitly specifies that a null image makes the ImageItem empty, and that a null image with a null label occupies no space. Alternate text applies when an existing image exceeds display capacity, not when the image is null. [Class description](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/ImageItem.html), [setImage contract](https://docs.oracle.com/javame/config/cldc/ref-impl/midp2.0/jsr118/javax/microedition/lcdui/ImageItem.html#setImage(javax.microedition.lcdui.Image)).
